tests/unit/algorithms/detection/adapters/mmdet/nncf/test_mmdet_nncf_builder.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2021-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import os
+import tempfile
+
+import numpy as np
+import torch
+from nncf.torch.nncf_network import NNCFNetwork
+
+from otx.algorithms.common.adapters.nncf.compression import NNCFMetaState
+from otx.algorithms.detection.adapters.mmdet.nncf.builder import build_nncf_detector
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+from tests.unit.algorithms.common.adapters.mmcv.nncf.test_helpers import (
+    create_config,
+    create_dataset,
+    create_model,
+)
+
+
+@e2e_pytest_unit
+def test_build_nncf_detector():
```

Comment:
I think this test function can be separated by each dependent test case (maybe 3?).
How about separating to keep one test for one case?

Reply:
This is not mine too, I just restored what I accidently deleted

Reply:
I see. but how about separating them at this time to make it more readable?

Reply:
I'm not sure, This function is test for one function "build_nncf_detector" and it just check some various situations. I don't want to refactor this whole test in this PR.

Reply:
Yes you're right. Refactoring whole test cases is impossible. I just want to test function a little clean step by step. If you don't have time for this, then I can do this. Could I add commit for this?

Reply:
Sure, why not?